### PR TITLE
remove useless if and update

### PR DESCRIPTION
**Pull request by rigazilla**
removed useless if (always true)
removed redundant  infinispan CR update
added error check on statefulset update

### CODE REVIEW

**Comment by dmvolod** [comment]
@rigazilla I also open PR #513 update related to unhandled errors. Please have a look. You can use code from it as a large number of places to be validate

**Comment by dmvolod** [comment]
Please also note, that handling errors in pod updates always fails
https://travis-ci.org/github/infinispan/infinispan-operator/jobs/715198324

**Comment by rigazilla** [comment]
> Please also note, that handling errors in pod updates always fails
> https://travis-ci.org/github/infinispan/infinispan-operator/jobs/715198324

this is surprising :) 
I would say we can remove that set

File: pkg/controller/infinispan/infinispan_controller.go [hunk 1]
```diff
@@ -1246,21 +1246,19 @@ func (r *ReconcileInfinispan) reconcileGracefulShutdown(ispn *infinispanv1.Infin
...
-				updateStatus = true
-				zeroReplicas := int32(0)
-				statefulSet.Spec.Replicas = &zeroReplicas
-				err = r.client.Update(context.TODO(), statefulSet)
-				if err != nil {
-					logger.Error(err, "failed to update StatefulSet", "StatefulSet.Name", statefulSet.Name)
-					return &reconcile.Result{}, err
-				}
+			ispn.Status.ReplicasWantedAtRestart = *statefulSet.Spec.Replicas
+			err := r.client.Status().Update(context.TODO(), ispn)
```
Comment (dmvolod) [inline pkg/controller/infinispan/infinispan_controller.go:1250]: `updateStatus` flag indicates that we need to update Status later in one place. Do we really need to update Status there

File: pkg/controller/infinispan/infinispan_controller.go [hunk 2]
```diff
@@ -1246,21 +1246,19 @@ func (r *ReconcileInfinispan) reconcileGracefulShutdown(ispn *infinispanv1.Infin
...
+			updateStatus = true
+			zeroReplicas := int32(0)
+			statefulSet.Spec.Replicas = &zeroReplicas
+			err = r.client.Update(context.TODO(), statefulSet)
+			if err != nil {
+				logger.Error(err, "failed to update StatefulSet", "StatefulSet.Name", statefulSet.Name)
+				return &reconcile.Result{}, err
 			}
 		}
 		if statefulSet.Status.CurrentReplicas == 0 {
```
Comment (dmvolod) [inline pkg/controller/infinispan/infinispan_controller.go:1264]: Next lines also contains code, which optimizer can ignore in some cases and didn't set a proper conditions
```
if statefulSet.Status.CurrentReplicas == 0 {
			updateStatus = updateStatus || ispn.SetCondition("gracefulShutdown", metav1.ConditionTrue, "")
			updateStatus = updateStatus || ispn.SetCondition("stopping", metav1.ConditionFalse, "")
}
```

File: pkg/controller/infinispan/infinispan_controller.go (hunk 1 above)
Comment (rigazilla) [reply pkg/controller/infinispan/infinispan_controller.go:1250]: We need to do it before zeroing the Replicas in the StatefulSet, after that the replicas needed at restart is lost if an error occurs in the Infinispan.Status update.
I've removed the `updateStatus=true` that is not needed

**Comment by rigazilla** [comment]
merged, thanks!

**Pull request merged**